cpp/src/gandiva/precompiled/decimal_wrapper.cc

```diff
@@ -0,0 +1,43 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "gandiva/precompiled/decimal_ops.h"
+#include "gandiva/precompiled/types.h"
+
+extern "C" {
+
+/// TODO : Passing around structs in IR can be fragile due to c-abi compatibility issues.
+/// This seems to work for now, but will need to revisit if we hit issues.
+struct DecimalSplit {
+  int64_t high_bits;
+  uint64_t low_bits;
+};
+
+FORCE_INLINE
```

Comment:
By curiosity, is this picked up by LLVM when compiling IR?
Generally, I would trust LLVM to inline small functions when a high optimization level is selected.

Reply:
early on when I was adding gandiva functions, I remember this made a difference. but, it's likely I didn't have all the optimizer passes working at that point. I'll test again and check if this is still required.